### PR TITLE
Bugfix/fix testconfig exit code

### PR DESCRIPTION
**Pull request by msiebeneicher**
## The purpose of this pull request

- fixed exit code for successfully config tests
- fixed doc example

## Steps to verify

```sh
gollum -tc config/socket_receive.conf && echo $?
```

## Checklist

- [x] `make test` executed successfully
- [ ] docs updated

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -77,11 +77,28 @@ func mainWithExitCode() int {
 	logrus.Debug("GOLLUM STARTING")
 	defer logrus.Debug("GOLLUM STOPPED")
 
-	config := readConfig()
```
Comment (arnecls) [inline main.go]: Why move all this testing and flag check code back to the main function?
The main function should not check for anything but boolean flags. Everything else should go into sub functions.

File: main.go [hunk 2]
```diff
@@ -179,17 +196,6 @@ func readConfig() *core.Config {
...
 
-	if testAndExit {
-		coordinator := NewCoordinator()
-		if err := coordinator.Configure(config); err != nil {
-			logrus.WithError(err).Error("Configure pass failed.")
-		} else {
-			fmt.Println("Config OK.")
-		}
-		coordinator.Shutdown()
-		return nil
```
Comment (arnecls) [inline main.go]: Why not simply exit here instead of returning and all that refactoring?

Comment (msiebeneicher) [reply main.go]: Because an exit in a readConfig() method just feels wrong. 
This PR is more a clean up instead of of refactoring in my point of view.

**Comment by arnecls** [comment]
I adjusted the config file retrieval part a bit.
Would be fine for me with that change.

**Pull request merged**